tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/stretchr/testify/require"
+
+	bv1 "github.com/rancher/backup-restore-operator/pkg/apis/resources.cattle.io/v1"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/pkg/config"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	backupName = namegen.AppendRandomString("backup")
+	secretName = namegen.AppendRandomString("broSecretName")
+	roleName   = namegen.AppendRandomString("broRole")
+	userList   = []*management.User{}
+	projList   = []*management.Project{}
+	roleList   = []*management.RoleTemplate{}
+
+	rules = []management.PolicyRule{
+		{
+			APIGroups: []string{"management.cattle.io"},
+			Resources: []string{"projects"},
+			Verbs:     []string{psaRole},
+		},
+	}
+)
+
+const (
+	backupEnabled        = true
+	backupChartNamespace = "cattle-resources-system"
+	backupChartName      = "rancher-backup"
+	backupSteveType      = "resources.cattle.io.backup"
+	restoreSteveType     = "resources.cattle.io.restore"
+	psaRole              = "updatepsa"
+	roleContext          = "cluster"
+	clusterProject       = "System"
+	cluster              = "local"
+	member               = "user"
+)
+
+func setBackupObject() bv1.Backup {
+	broConfig := new(charts.Config)
+	config.LoadConfig(charts.ConfigurationFileKey, broConfig)
+
+	backup := bv1.Backup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: backupName,
+		},
+		Spec: bv1.BackupSpec{
+			ResourceSetName: broConfig.ResourceSetName,
+		},
+	}
+	return backup
+}
+
+func setRestoreObject() bv1.Restore {
+	broConfig := new(charts.Config)
+	config.LoadConfig(charts.ConfigurationFileKey, broConfig)
+
+	restore := bv1.Restore{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "restore-",
+		},
+		Spec: bv1.RestoreSpec{
+			BackupFilename: backupName,
+			Prune:          &broConfig.Prune,
+		},
+	}
+	return restore
+}
+
+func installBroChart(t *testing.T, client *rancher.Client, chartInstallOptions *charts.InstallOptions, chartFeatureOptions *charts.RancherBackupOpts, project *management.Project) {
+	_, err := createOpaqueS3Secret(client.Steve)
+	require.NoError(t, err)
+
+	logrus.Info("Installing backup chart")
+	err = charts.InstallRancherBackupChart(client, chartInstallOptions, chartFeatureOptions, true)
+	require.NoError(t, err)
+
+	logrus.Info("Waiting backup chart deployments to have expected number of available replicas")
+	err = charts.WatchAndWaitDeployments(client, project.ClusterID, backupChartNamespace, metav1.ListOptions{})
+	require.NoError(t, err)
+
+	logrus.Info("Waiting backup chart DaemonSets to have expected number of available nodes")
+	err = charts.WatchAndWaitDaemonSets(client, project.ClusterID, backupChartNamespace, metav1.ListOptions{})
+	require.NoError(t, err)
+
+	logrus.Info("Waiting backup chart StatefulSets to have expected number of ready replicas")
```

Comment:
very small nit, I think the word "for" is missing in the waiting log statements.